sdk/documentintelligence/ai-document-intelligence-rest/src/documentIntelligence.ts

```diff
@@ -63,5 +65,39 @@ export default function createClient(
     },
   });
 
+  client.pipeline.addPolicy({
+    name: "ResponseStreamToUint8ArrayPolicy",
+    sendRequest: async (req, next) => {
+      logger.info("ResponseStreamToUint8ArrayPolicy starts");
+
+      const figuresUrlPattern = /\/documentModels\/[^/]+\/analyzeResults\/[^/]+\/figures\/[^/]+/;
+      if (req.method === "GET" && figuresUrlPattern.test(req.url)) {
+        logger.info(`Matched URL pattern for figures: ${req.url}`);
+        req.streamResponseStatusCodes = new Set([200]);
+        logger.info("Set streamResponseStatusCodes to 200 for the figures API");
+      }
+
+      const response = await next(req);
+
+      if (req.method === "GET" && figuresUrlPattern.test(req.url)) {
+        logger.info(`Handling response for figures URL: ${req.url}`);
+        if (!response.readableStreamBody) {
+          logger.error("Expected response to have a readable stream but found none");
+          throw new RestError("Expected response to have a readable stream", {
+            code: RestError.PARSE_ERROR,
+          });
+        }
+        logger.info("Converting readable stream to Uint8Array");
+        // Casting Uint8Array as NodeJS.ReadableStream which gets passed on as response.body, this lets users consume Uint8Array response body directly
+        response.readableStreamBody = (await streamToUint8Array(
+          response.readableStreamBody,
+        )) as unknown as NodeJS.ReadableStream;
```

Comment:
more context..

This transformation and casting is needed because 
- The get method is implemented in core-client-rest, and it's response cannot be tweaked from the document-intelligence SDK. 
- Leaving us to only play at the policy level.
- We cannot modify the body attribute in the policy because it will be overridden, leading me to play only with `readableStreamBody` attribute
![image](https://github.com/user-attachments/assets/a2ee5996-addc-4334-b10d-bbcdb05ee1e4)